### PR TITLE
Make it possible to get RFC2254 escapes when print ASN1_STRING

### PR DESCRIPTION
**Pull request by levitte**

### CODE REVIEW

**Comment by levitte** [comment]
RT ticket: https://rt.openssl.org/Ticket/Display.html?id=1466

**Comment by levitte** [comment]
I intend to add this change to both master and 1.0.2

File: include/openssl/asn1.h [hunk 1]
```diff
@@ -410,6 +410,11 @@ typedef const ASN1_ITEM *ASN1_ITEM_EXP (void);
 # define ASN1_STRFLGS_DUMP_DER           0x200
 
 /*
+ * This flag determines to do RC2254 escaping.
```
Comment (richsalz) [inline include/openssl/asn1.h]: determines --> specifies

**Comment by richsalz** [comment]
minor comment wording nit.
+1 ship it.

File: include/openssl/asn1.h (hunk 1 above)
Comment (levitte) [inline include/openssl/asn1.h]: Will this do?

```
 * This flag specifies that RC2254 escaping should be performed.
```

Comment (richsalz) [inline include/openssl/asn1.h]: Yes.
Re-confirm =1

**Comment by levitte** [comment]
Merged

**Comment by levitte** [comment]
.. and backported to 1.0.2

**Comment by mattcaswell** [comment]
Should this be reverted as a feature?

**Comment by levitte** [comment]
Do you mean:

-   Revert from 1.0.2?
-   Revert from both master and 1.0.2?

I've just create a revert PR for 1.0.2, in #1095

**Comment by mattcaswell** [comment]
Certainly from 1.0.2. I'll leave it to you and @richsalz to determine whether it is appropriate for 1.1.0 or not.

**Pull request closed without merging**